README.md

```diff
@@ -17,59 +21,495 @@ kubectl apply -k 'github.com/howardjohn/service-apis-demo'
+## Gateway Tutorial
+
+The following tutorial explores different resources of the Gateway API. It goes through four different examples of Gateway that have increasing complexity to demonstrate aspects of the API.
+
+- [Single Service Gateway](./single-service-gateway) shows the simplest possible Gateway which matches all traffic and sends it to a single Service.
+- [Multi-Service Gateway](./multi-service-gateway) shows a very typical Gateway deployment where an HTTPRoute with multiple routing rules matches and routes traffic to a group of different Services.
+- [Multi-Route Gateway](./multi-route-gateway) shows how a Gateway can be shared by multiple HTTPRoute resources to enable shared infrastructure between many independent groups of users.
+- [Traffic Split](./traffic-split) demonstrates how traffic can be gracefully split between multiple Services to achieve blue-green traffic deployments.
+
+Pull this repo to your local environment. All commands should be run from the `/service-apis-demo` folder. Before starting the tutorial install the demo application and gateway-demo Namespace in your cluster. Switch your context to the gateway-demo Namespace for easier configuration and inspection.
+
+```bash
+$ kubectl apply -f demo-app.yaml
+namespace/gateway-demo created
+deployment.apps/foo created
+service/foo-svc created
+deployment.apps/bar created
+service/bar-svc created
+deployment.apps/l7-default-backend created
+service/default-backend created
+
+$ kubectl config set-context --current --namespace=gateway-demo
+Context "gke-us-east" modified.
 ```
 
-Send a request:
-```shell
-# HTTP, to Gateway
-curl http://$INGRESS_HOST:$INGRESS_PORT/get -H "Host: gateway.local"
-# HTTPS, to Gateway
-curl --resolve gateway.local:$INGRESS_PORT_HTTPS:$INGRESS_HOST https://gateway.local:$INGRESS_PORT_HTTPS/get -k
-# HTTP, to Ingress
-curl http://$INGRESS_HOST:$INGRESS_PORT/get -H "Host: ingress.local"
+You're now ready to begin the tutorial!
+
+### GatewayClass
+
+The `GatewayClass` is a resource that defines a template for L4 and L7 load balancers in a Kubernetes cluster. `GatewayClasses` are instantiated as `Gateways` which represent running load balancer instances. `Gateways` can be implemented by different controllers, often with different configuration. Each `Gateway` should specify a class, a reference to a `GatewayClass` resource that contains additional configuration including the name of the controller that should implement the class. `GatewayClass` resources contain an optional parameters field. This can be used to reference additional configuration for this class.
+
+1. The following `istio GatewayClass` allows Gateways of this class type to be created in the cluster. Note that the `GatewayClass` exists in the cluster Namespace and thus could be deployed as a Gateway in any Namespace (if RBAC rules permit it). 
+
+```yaml
+apiVersion: networking.x-k8s.io/v1alpha1
+kind: GatewayClass
+metadata:
+  name: istio
```

Comment:
Do we want to make this name istio specific? Or something implementation-generic?